### PR TITLE
Upgrade to VS 2017

### PR DESCRIPTION
**Pull request by natemcmaster**
WIP: still needs to be verified in VS 2017

cref aspnet/Universe#469

### CODE REVIEW

**Comment by natemcmaster** [comment]
@ryanbrandenburg it looks like tests are failing. I think it's due to a temporary bug in vstest which runs the tests in the wrong working directory. https://github.com/Microsoft/vstest/issues/311 This can usually be worked around by using AppContext.BaseDirectory (netcoreapp) or AppDomain.CurrentDirectory.BaseDirectory (net451) to find test files

File: build/common.props [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<Project>
+  <Import Project="..\version.props" />
```
Comment (bricelam) [inline build/common.props]: This needs `$(MSBuildThisFileDirectory)`. Includes are always relative to the "entry" project file.

Comment (natemcmaster) [inline build/common.props]: Not quite. Relative paths in imported projects are interpreted relative to the directory of the importing project.

Comment (natemcmaster) [inline build/common.props]: ...but, the relative path of Import Project is relative to the file in which its expressed.

Or so I've been led to believe. This works in all of our other repos

Comment (bricelam) [inline build/common.props]: Weird. I'm getting assembly version 1.0.0.0 without it.

Comment (natemcmaster) [inline build/common.props]: Building this repo? It seems to work for me

image

File: appveyor.yml [hunk 2]
```diff
@@ -10,4 +10,5 @@ build_script:
   - build.cmd --quiet verify
 clone_depth: 1
 test: off
-deploy: off
\ No newline at end of file
+deploy: off
+os: Visual Studio 2017 RC
```
Comment (pranavkm) [inline appveyor.yml]: Do we need to do this in all our repos?

File: test/LocalizationWebsite/LocalizationWebsite.csproj [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <Import Project="..\..\build\common.props" />
+
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp1.1;net451</TargetFrameworks>
+    <!-- TODO remove when https://github.com/dotnet/sdk/issues/396 is resolved -->
+    <RuntimeIdentifier Condition=" '$(TargetFramework)' != 'netcoreapp1.1' ">win7-x64</RuntimeIdentifier>
+    <OutputType>Exe</OutputType>
+    <!-- workaround issues with satellite assembly generation and -->
```
Comment (pranavkm) [inline test/LocalizationWebsite/LocalizationWebsite.csproj]: and signing?

File: build/common.props (hunk 1 above)
Comment (bricelam) [inline build/common.props]: Building from VS?

Comment (bricelam) [inline build/common.props]: > this repo?
No, EF, but it's the same pattern.

File: test/LocalizationWebsite/LocalizationWebsite.csproj (hunk 3 above)
Comment (natemcmaster) [inline test/LocalizationWebsite/LocalizationWebsite.csproj]: yes, that was the rest of the sentence. @ryanbrandenburg have you already opened a bug for this, or should I?

File: appveyor.yml (hunk 2 above)
Comment (natemcmaster) [inline appveyor.yml]: Yes. The default image doesn't have .NET Framework 4.6, which is required for dotnet-test

File: build/common.props (hunk 1 above)
Comment (natemcmaster) [inline build/common.props]: Yeah, works in VS and command line. Something else must be going on since the import would fail if it couldn't find the file.

Comment (bricelam) [inline build/common.props]: Huh, good point. 😄 Yuck. Not something I want to debug...

Comment (natemcmaster) [inline build/common.props]: https://github.com/KirillOsenkov/MSBuildStructuredLog

Comment (natemcmaster) [inline build/common.props]: and may the force be with you

File: test/LocalizationWebsite/LocalizationWebsite.csproj (hunk 3 above)
Comment (natemcmaster) [inline test/LocalizationWebsite/LocalizationWebsite.csproj]: Nvm, I verified this works on 1.0.0-rc4-004755. Will update the TODO here

**Comment by natemcmaster** [comment]
superseded by #322

**Pull request closed without merging**